src/components/Modals/SemesterMenu.vue

```diff
@@ -45,7 +64,7 @@
 import { defineComponent } from 'vue';
 
 export default defineComponent({
-  emits: ['open-delete-semester-modal', 'open-edit-semester-modal'],
+  emits: ['open-delete-semester-modal', 'open-edit-semester-modal', 'open-clear-semester-modal'],
   data() {
     return {
       isOpenModal: false,
```

Comment:
I'm not sure if this data field is really needed--isDeleteSemesterOpen, isEditSemesterOpen, and isClearSemesterOpen are all stored in the data of the Semester component, so isOpenModal could be passed into SemesterMenu as as a prop such that isOpenModal = isDeleteSemesterOpen || isEditSemesterOpen || isClearSemesterOpen. It's probably best to minimize use of mutable state.